bundles/org.openhab.binding.bsblan/src/main/java/org/openhab/binding/bsblan/internal/BsbLanHandlerFactory.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.bsblan.internal;
+
+import static org.openhab.binding.bsblan.internal.BsbLanBindingConstants.*;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.openhab.binding.bsblan.internal.handler.BsbLanBridgeHandler;
+import org.openhab.binding.bsblan.internal.handler.BsbLanParameterHandler;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * The {@link BsbLanHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Peter Schraffl - Initial contribution
+ */
+@Component(service = ThingHandlerFactory.class, configurationPid = "binding.bsblan")
+public class BsbLanHandlerFactory extends BaseThingHandlerFactory {

```

Comment:
Please add `@NonNullByDefault`

Reply:
Really? This is causing a compilation error and other bindings don't have this either on their handler factory class.

```[ERROR] ...\BsbLanHandlerFactory.java:[56,1988] The default '@NonNull' conflicts with the inherited '@Nullable' annotation in the overridden method from BaseThingHandlerFactory```